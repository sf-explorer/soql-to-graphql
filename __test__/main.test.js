var converter = require('../build/src/main.js').default;

describe('converter function', () => {


  const simpleQuery = "select Id, Name, Owner.Name, (select Id from Opportunities) from Account where Name like '%N' limit 3"
  it('Simple Query: ' + simpleQuery, () => {

    expect(converter(simpleQuery)).toMatchSnapshot();
  });

  const queryWithVar = "select Id, Name from Account where (Id = :recordId and Name like :name) or BillingCountry like :country limit 3"

  it('Query with var ' + queryWithVar, () => {

    expect(converter(queryWithVar, { recordId: 'ID!', name: 'String = "%"', country: 'String' })).toMatchSnapshot();
  });



  const query = `SELECT
  Id,
  Name,
  Address__c,
  City__c,
  State__c,
  Description__c,
  Price__c,
  Baths__c,
  Beds__c,
  Thumbnail__c,
  Location__Latitude__s,
  Location__Longitude__s
FROM Property__c
WHERE
  (Name LIKE :searchPattern
  OR City__c LIKE :searchPattern
  OR Tags__c LIKE :searchPattern)
  AND Price__c <= :safeMaxPrice
  AND Beds__c > :safeMinBedrooms
  AND Baths__c >= :safeMinBathrooms
WITH USER_MODE
ORDER BY Price__c desc
LIMIT 3
OFFSET 3
`

  it('Advanced conversion test', () => {

    expect(converter(query, {
      searchPattern: 'String ="%"',
      safeMaxPrice: 'Currency = 10000',
      safeMinBedrooms: 'Double = 3',
      cursorSize: 'Double = 3',
      safeMinBathrooms: 'Double = 2',
    })).toMatchSnapshot()
  })

  const inQuery = "select Id, Name, Owner.Name from Account where Id in ('0030700000ywKMfAAM') limit 3"
  it('in Query: ' + inQuery, () => {

    expect(converter(inQuery)).toMatchSnapshot()
  })

  const subWhere = "select Id FROM OrderItem WHERE Order.AccountId = :accountId limit 3"
  it('Sub where Query: ' + subWhere, () => {

    expect(converter(subWhere)).toMatchSnapshot()
  })

  

})