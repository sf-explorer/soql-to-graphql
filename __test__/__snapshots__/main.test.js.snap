// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter function Advanced conversion test 1`] = `
"query ($searchPattern: String ="%", $safeMaxPrice: Currency = 10000, $safeMinBedrooms: Double = 3, $cursorSize: Double = 3, $safeMinBathrooms: Double = 2) {
    uiapi {
        query {
            Property__c (first: 3, where: {or: [{Name: {like: $searchPattern}}, {or: [{City__c: {like: $searchPattern}}, {and: [{Tags__c: {like: $searchPattern}}, {and: [{Price__c: {lte: $safeMaxPrice}}, {and: [{Beds__c: {gt: $safeMinBedrooms}}, {Baths__c: {gte: $safeMinBathrooms}}]}]}]}]}]}, orderBy: {Price__c: {order: DESC}}) {
                edges {
                    node {
                        Id
                        Name {
                            value
                        }
                        Address__c {
                            value
                        }
                        City__c {
                            value
                        }
                        State__c {
                            value
                        }
                        Description__c {
                            value
                        }
                        Price__c {
                            value
                        }
                        Baths__c {
                            value
                        }
                        Beds__c {
                            value
                        }
                        Thumbnail__c {
                            value
                        }
                        Location__Latitude__s {
                            value
                        }
                        Location__Longitude__s {
                            value
                        }
                    }
                }
                pageInfo {
                    endCursor
                    hasNextPage
                    hasPreviousPage
                }
            }
        }
    }
}"
`;

exports[`converter function Query with var select Id, Name from Account where (Id = :recordId and Name like :name) or BillingCountry like :country limit 3 1`] = `
"query ($recordId: ID!, $name: String = "%", $country: String) {
    uiapi {
        query {
            Account (first: 3, where: {and: [{Id: {eq: $recordId}}, {or: [{Name: {like: $name}}, {BillingCountry: {like: $country}}]}]}) {
                edges {
                    node {
                        Id
                        Name {
                            value
                        }
                    }
                }
                pageInfo {
                    endCursor
                    hasNextPage
                    hasPreviousPage
                }
            }
        }
    }
}"
`;

exports[`converter function Simple Query: select Id, Name, Owner.Name, (select Id from Opportunities) from Account where Name like '%N' limit 3 1`] = `
"query {
    uiapi {
        query {
            Account (first: 3, where: {Name: {like: "%N"}}) {
                edges {
                    node {
                        Id
                        Name {
                            value
                        }
                        Owner {
                            Name {
                                value
                            }
                        }
                        Opportunities {
                            edges {
                                node {
                                    Id
                                }
                            }
                        }
                    }
                }
                pageInfo {
                    endCursor
                    hasNextPage
                    hasPreviousPage
                }
            }
        }
    }
}"
`;

exports[`converter function Sub where Query: select Id FROM OrderItem WHERE Order.AccountId = :accountId limit 3 1`] = `
"query {
    uiapi {
        query {
            OrderItem (first: 3, where: {Order: {AccountId: {eq: "accountId"}}}) {
                edges {
                    node {
                        Id
                    }
                }
                pageInfo {
                    endCursor
                    hasNextPage
                    hasPreviousPage
                }
            }
        }
    }
}"
`;

exports[`converter function Where boolean: Select IsClosed from Case where IsClosed=true 1`] = `
"query {
    uiapi {
        query {
            Case (where: {IsClosed: {eq: true}}) {
                edges {
                    node {
                        IsClosed {
                            value
                        }
                    }
                }
            }
        }
    }
}"
`;

exports[`converter function in Query: select Id, Name, Owner.Name from Account where Id in ('0030700000ywKMfAAM') limit 3 1`] = `
"query {
    uiapi {
        query {
            Account (first: 3, where: {Id: {in: ["0030700000ywKMfAAM"]}}) {
                edges {
                    node {
                        Id
                        Name {
                            value
                        }
                        Owner {
                            Name {
                                value
                            }
                        }
                    }
                }
                pageInfo {
                    endCursor
                    hasNextPage
                    hasPreviousPage
                }
            }
        }
    }
}"
`;
